components/merkledb/src/backends/rocksdb.rs

```diff
@@ -152,6 +153,17 @@ impl Database for RocksDB {
         w_opts.set_sync(true);
         self.do_merge(patch, &w_opts)
     }
+
+    fn create_checkpoint(&self, path: &Path) -> crate::Result<Box<dyn Database>> {
+        let checkpoint = Checkpoint::new(&*self.db)?;
+        checkpoint.create_checkpoint(path)?;
+        let options = DbOptions {
```

Comment:
Why the options are not preserved? I can understand `create_if_missing: false`, but can't understand why other options are default.

Reply:
This will, unfortunately, break object safety for the `Database` trait.

Reply:
Not applicable to a new version.